Added `deploy_contract` docs (#1840)
software-mansion/protostar: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/deploy_contract_cairo.md

> # `deploy_contract_cairo0`

Reviewer: why is the `0` missing from the file name?

Author: https://github.com/software-mansion/protostar/pull/1840/commits/9984117aeeb36097db629c0af69223c212c4cb91

```cairo
fn deploy_contract_cairo0(contract: felt252, calldata: Array::<felt252>) -> 
Result::<felt252, RevertedTransaction>
```

> Declares and deploys a cairo0 contract given its name defined in the [protostar.toml](../../04-configuration-file.md) configuration

Reviewer 2: ```suggestion
Declares and deploys a cairo 0 contract given its name defined in the [protostar.toml](../../04-configuration-file.md) configuration
```

Author: https://github.com/software-mansion/protostar/pull/1840/commits/8534705c8eb65ab65d4d5399e8e600b7388f3316

file.

```cairo title="Example"
use result::ResultTrait;
use array::ArrayTrait;

fn my_function() {

    let mut calldata = ArrayTrait::new();
    calldata.append(10);
    calldata.append(11);
    calldata.append(12);

    let address = deploy_contract_cairo0('mycontract', calldata).unwrap();

    // ...
}
```

Errors raised by the constructor can be handled in the same way as for [deploy](deploy.md)

:::info

> `deploy_contract_cairo0` is just a syntactic sugar over executing cheatcodes `declare_cairo0` -> `prepare` -> `deploy` separately, and it's what it does under the hood.

Reviewer 2: It's not really a syntactic sugar 
```suggestion
`deploy_contract_cairo0` is just a function, which chains cheatcodes `declare_cairo0` -> `prepare` -> `deploy` under the hood.
```

Author: calls instead of chains, as those are not techincally chained
